authorization/invitation/service/invitation_service.go

```diff
@@ -151,6 +166,40 @@ func (s *invitationServiceImpl) Issue(ctx context.Context, issuingUserId uuid.UU
 					return errors.NewInternalError(ctx, error)
 				}
 			}
+
+			notifications = append(notifications, invitationNotification{
+				invitation: inv,
+				roles:      invitation.Roles,
+			})
+
+		}
+
+		inviter, err := s.Repositories().Identities().Load(ctx, issuingUserId)
+		if err != nil {
+			return err
+		}
+
+		// Use the notification service to send invitation e-mails to the invited users, in a separate thread
+		// Currently we only support two types of invitations;
+		//
+		// 1) Invite user to team, membership only, no organization
+		// 2) Invite user to space, roles only, no organization
+		//
+		if inviteToIdentity != nil {
+			identityResource, err := s.Repositories().ResourceRepository().Load(ctx, inviteToIdentity.IdentityResourceID.String)
+			if err != nil {
+				return err
+			}
+
+			if identityResource.ResourceType.Name == authorization.IdentityResourceTypeTeam {
+				err = s.processTeamInviteNotifications(ctx, inviteToIdentity, inviter.User.FullName, notifications, witURL)
+			}
+		} else if inviteToResource != nil && inviteToResource.ResourceType.Name == authorization.ResourceTypeSpace {
+			err = s.processSpaceInviteNotifications(ctx, inviteToResource, inviter.User.FullName, notifications, witURL)
+		}
+
+		if err != nil {
```

Comment:
I guess we should check if we didn't send any notifications (invateToIdentity == nil & inviteToResource == nil or unknown/unsupported resource type (org or whatever)) and return NotFoundError in that case.

Reply:
I don't think we should error here, while we only support sending _notifications_ for two particular 'types' of invitations, it should still be possible to send other invitation types.  I've updated the comment to reflect this.

Reply:
Sure but for now we support only these two types. I don't think we should silently ignore other types. Let's return an error if we can't actually sent any invitation because we don't support it.